google-cloud-storage/lib/google/cloud/storage/service.rb

```diff
@@ -48,10 +48,12 @@ def initialize project, credentials, retries: nil, timeout: nil
           @service.client_options.open_timeout_sec = timeout
           @service.client_options.read_timeout_sec = timeout
           @service.client_options.send_timeout_sec = timeout
+          @service.client_options.transparent_gzip_decompression = false
           @service.request_options.retries = retries || 3
           @service.request_options.header ||= {}
           @service.request_options.header["x-goog-api-client"] = \
             "gl-ruby/#{RUBY_VERSION} gccl/#{Google::Cloud::Storage::VERSION}"
+          @service.request_options.header["Accept-Encoding"] = "gzip"
```

Comment:
Correct: The decompressive transcoding in the Storage service is no longer needed, since the client library will now decompress data sent with the `Content-Encoding: gzip` response header. The decompressive transcoding in the service is also not *wanted*, since it creates the situation in which downloaded data cannot be verified using the hash of the same data in its compressed state.

Reply:
Ah okay, that makes sense. Not allowing decompressive transcoding also has the advantage of a user not incurring the cost from downloading the decompressed file size. Thanks for clarifying!